pkg/logs/message/message.go

```diff
@@ -13,6 +13,13 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util"
 )
 
+// Payload represents an encoded collection of messages ready to be sent to the intake
+type Payload struct {
+	Messages []*Message
```

Comment:
Yep we retain the messages for the auditor to maintain context at the end of the pipeline. 
this could probably be optimized by only storing the metadata of the last message. However i've benchmarked this and it has little impact on real workloads since our in-agent buffers are very small. 

Could be worth a followup though if we ever decide to make the buffers configurable (which has been discussed)